x-pack/plugins/alerting/server/application/backfill/methods/delete/delete_backfill.ts

```diff
@@ -85,7 +85,7 @@ async function deleteWithOCC(context: RulesClientContext, { id }: { id: string }
     );
 
     // remove the associated task
-    context.taskManager.removeIfExists(id);
+    await context.taskManager.removeIfExists(id);
```

Comment:
My understanding of this method is that it removes an entity, _backfill_. And that entity has a task associated that needs to be removed.

That's why removing the task felt like part of the business logic and needs to be awaited and addressed if failed. Accepting your changes as it looks like I misunderstood it 😇

Reply:
@ymao1 thoughts?  I think when I saw this I was thinking that we'd want to do this async rather than sync, since the method is async.  But also re-thinking about zombie tasks :-).  Should we change this back to Alejandro's `await` instead?

Reply:
If you don't mind, I'd rather leave it as you suggested for this PR (unattended, but caught) because it doesn't change the business logic that we currently have. And have the potential enhancement discussion somewhere else.

Keeping this PR up-to-date and without conflicts is a real challenge 😇

Reply:
Yea, we can address in a followup.

Reply:
I opened https://github.com/elastic/kibana/issues/182208 to track